### PR TITLE
openstack integration with ceph

### PR DESCRIPTION
**Pull request by hmaheswa**
integrating ceph rgw with osp17 keystone.
refer these documents:
https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/17.0/html/standalone_deployment_guide/index 
https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/5/html/object_gateway_guide/advanced-configuration#ceph-object-gateway-and-openstack-keystone

pass logs:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-SYIRFT/

Signed-off-by: Hemanth Maheswarla <hmaheswa@hmaheswa.remote.csb>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/misc_env/exec.py [hunk 1]
```diff
@@ -152,6 +154,34 @@ def get_systemctl_service_name(node, service_name, command):
...
+        f"OS_AUTH_URL={auth['auth_url']}",
+        f"OS_IDENTITY_API_VERSION={clouds_yaml['clouds']['standalone']['identity_api_version']}",
+        "export OS_USERNAME OS_PASSWORD OS_PROJECT_NAME OS_USER_DOMAIN_NAME OS_PROJECT_DOMAIN_NAME"
+        + " OS_AUTH_URL OS_IDENTITY_API_VERSION",
+    ]
+    cmd = ";".join(commands)
+    return cmd
+
+
 def run(ceph_cluster, **kwargs: Any) -> int:
```
Comment (TejasC88) [inline tests/misc_env/exec.py:185]: The exec.py file is used by a lot of scripts , can you attach pass log of 1 or 2 more tests which use this py file ?

Comment (hmaheswa) [reply tests/misc_env/exec.py:185]: sure @TejasC88 , will provide more logs

Comment (hmaheswa) [reply tests/misc_env/exec.py]: @TejasC88 , reverted back exec.py changes. created ~/os_auth.py file for openstack authentication on osp node.

File: tests/misc_env/install_standalone_osp.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
+import ipaddress
```
Comment (sunilkumarn417) [inline tests/misc_env/install_standalone_osp.py:1]: Hi @hmaheswa , can we modularize the OSP installation, so we could enhance anytime with newer requirements.

File: conf/pacific/rgw/ceph_cluster+standalone_osp_node.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        no-of-volumes: 4
+        role:
+          - osd
+      node5:
+        role:
+          - rgw
+      node6:
+        role:
+          - client
+  - ceph-cluster:
```
Comment (sunilkumarn417) [inline conf/pacific/rgw/ceph_cluster+standalone_osp_node.yaml:37]: would it be nice, if we keep OSP conf separately and use with any configuration in future.

File: tests/misc_env/install_standalone_osp.py (hunk 2 above)
Comment (hmaheswa) [reply tests/misc_env/install_standalone_osp.py:1]: working on it @sunilkumarn417

**Comment by hmaheswa** [comment]
closing this PR to open below PR with modularized code:
https://github.com/red-hat-storage/cephci/pull/2100

**Pull request closed without merging**